static/app/components/tabs/index.spec.tsx

```diff
@@ -0,0 +1,203 @@
+            <Item key={tab.key}>{tab.label}</Item>
+          ))}
+        </TabList>
+        <TabPanels>
+          {TABS.map(tab => (
+            <Item key={tab.key}>{tab.content}</Item>
+          ))}
+        </TabPanels>
+      </Tabs>
+    );
+
+    // Click on the Activity tab
+    userEvent.click(screen.getByRole('tab', {name: 'Activity'}));
+
+    // The Activity tab is selected and its content rendered
+    expect(screen.getByRole('tab', {name: 'Activity'})).toHaveAttribute(
+      'aria-selected',
+      'true'
+    );
+    expect(screen.getByText(TABS[1].content)).toBeInTheDocument();
+  });
+
+  it('changes tabs on key press', () => {
+    render(
+      <Tabs>
+        <TabList>
+          {TABS.map(tab => (
+            <Item key={tab.key}>{tab.label}</Item>
+          ))}
+        </TabList>
+        <TabPanels>
+          {TABS.map(tab => (
+            <Item key={tab.key}>{tab.content}</Item>
+          ))}
+        </TabPanels>
+      </Tabs>
+    );
+
+    // Focus on tab list
+    userEvent.tab();
+    expect(screen.getByRole('tab', {name: 'Details'})).toHaveFocus();
+
+    // Press Arrow Right to select the next tab to the right (Activity)
+    userEvent.keyboard('{arrowRight}');
+
+    // The Activity tab is selected and its contents rendered
+    expect(screen.getByRole('tab', {name: 'Activity'})).toHaveAttribute(
+      'aria-selected',
+      'true'
+    );
+    expect(screen.getByText(TABS[1].content)).toBeInTheDocument();
+  });
+
+  it('changes tabs on key press in vertical orientation', () => {
+    render(
+      <Tabs orientation="vertical">
+        <TabList>
+          {TABS.map(tab => (
+            <Item key={tab.key}>{tab.label}</Item>
+          ))}
+        </TabList>
+        <TabPanels>
+          {TABS.map(tab => (
+            <Item key={tab.key}>{tab.content}</Item>
+          ))}
+        </TabPanels>
+      </Tabs>
+    );
+
+    // Focus on tab list
+    userEvent.tab();
+    expect(screen.getByRole('tab', {name: 'Details'})).toHaveFocus();
+
+    // Press Arrow Right to select the next tab below (Activity)
+    userEvent.keyboard('{arrowDown}');
+
+    // The Activity tab should now be selected and its contents rendered
+    expect(screen.getByRole('tab', {name: 'Activity'})).toHaveAttribute(
+      'aria-selected',
+      'true'
+    );
+    expect(screen.getByText(TABS[1].content)).toBeInTheDocument();
+  });
+
+  it('renders disabled tabs', () => {
+    render(
+      <Tabs>
+        <TabList>
+          {TABS.map(tab => (
+            <Item key={tab.key} disabled>
+              {tab.label}
+            </Item>
+          ))}
+        </TabList>
+        <TabPanels>
+          {TABS.map(tab => (
+            <Item key={tab.key}>{tab.content}</Item>
+          ))}
+        </TabPanels>
+      </Tabs>
+    );
+
+    TABS.forEach(tab => {
+      expect(screen.getByRole('tab', {name: tab.label})).toHaveAttribute(
+        'aria-disabled',
+        'true'
+      );
+    });
+  });
+});
```

Comment:
Would be nice to have some overflow behavior tests

Reply:
I thought about that too, but apparently RTL/`react-dom` doesn't support changing viewport sizes(?) Tell me if there's another way to test it